src/app/shared/dso-selector/modal-wrappers/claim-item-selector/claim-item-selector.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { BehaviorSubject } from 'rxjs';
+import { PaginatedList } from '../../../../core/data/paginated-list.model';
+import { RemoteData } from '../../../../core/data/remote-data';
+import { Item } from '../../../../core/shared/item.model';
+import { SearchResult } from '../../../search/models/search-result.model';
+import { DSOSelectorModalWrapperComponent } from '../dso-selector-modal-wrapper.component';
+import { getItemPageRoute } from '../../../../item-page/item-page-routing-paths';
+import { EPerson } from '../../../../core/eperson/models/eperson.model';
+import { DSpaceObject } from '../../../../core/shared/dspace-object.model';
+import { ViewMode } from '../../../../core/shared/view-mode.model';
+import { ProfileClaimService } from '../../../../profile-page/profile-claim/profile-claim.service';
+import { CollectionElementLinkType } from '../../../object-collection/collection-element-link.type';
+
+
+
+@Component({
+  selector: 'ds-claim-item-selector',
+  templateUrl: './claim-item-selector.component.html'
+})
+export class ClaimItemSelectorComponent extends DSOSelectorModalWrapperComponent implements OnInit {
```

Comment:
This new component needs TypeDocs to describe it.  All methods also need TypeDocs.